### PR TITLE
Refactor(core) data-models, shared and workflow packages

### PR DESCRIPTION
**Pull request by chrismclarke**
PR Checklist

- [x] PR title descriptive (can be used in release notes)

## Description
As part of work on #1695 it was noted that packages within the platform cannot be built independently due to circular dependency issues (mostly between scripts and data-models/workflows). 

It has also been previously noted that whilst workflows were originally designed so that each deployment could specify their own, pretty much all workflows up till now have been generalised for use across all deployments, and do not make much sense to be contained within the data-models workspace.

This PR solves these issues by:

- Move workflows out of the data-models folder to their own standalone package
- Move most script utils to shared utils folder
- Remove circular dependencies
- Minor fixes to tests flagged during type-checking
- Reduce number of workspace imports between packages

**Additional Changes**
- Improve way angular app imports code from shared workspaces
- Add readme documentation to data-models and shared workspaces

## Dev Notes
After the changes the frontend code was having issue importing from packages/data-models, due to the way the code was shared with the angular compiler. There was a bit of a disconnect from the way the frontend app would try import some files as ts and others compiled js (depending on how the related workspace package.json was defined and angular's own tsconfig files).

I think I've now found a cleaner solution to always import as .ts files (see implementation notes in data-models and shared workspace readme). As a result of this we know longer need additional scripts to compile/serve shared workspaces (have deprecated for now and removed from CI build - fun fact, this was what got the ball rolling for all the CI refactor in #2025), where the frontend code _should_ live-reload when any code changes happen in the `shared` or `data-models` packages

## Review Notes
These code changes mostly impact workflow scripts run, however also have a knock-on for the main app via code imported from `data-models` and `shared` workspaces.

I've manually tested that `deployment set` and `sync` workflows are still running as expected, and all script and shared workspace tests still pass. There is a possibility that some scripts may still fail to run if the imports have not been updated correctly, however lint and test suggests that everything is as expected.

I've also tested running frontend code and fixed immediate issues raised. The 'data-models' package code includes database definitions, which would throw error on start if implemented incorrectly (obvious fail). The 'shared' package is used only for data item processing, so needs testing with a template that uses `begin_items`. I've manually tested against the app-debug-content repo and all appears working correct. A second quick functional test would be appreciated. 

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

### CODE REVIEW

**Review by jfmcquade — APPROVED** [review]
This looks good. It's a little hard to follow all the file changes, but the PR description makes sense, as do the READMEs and comments throughout.

The live reload seems to work for changes to files in the `data-models` and `shared` folders.

File: tsconfig.app.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
   "include": [
     "src/**/*.d.ts",
     "src/app/shared/components/template/components/odk-form/libs/**/*.js",
-    "src/app/shared/components/template/components/odk-form/libs/**/*.ts"
+    "src/app/shared/components/template/components/odk-form/libs/**/*.ts",
+    // Ensure any ts-based packages used by frontend app are listed here
+    // to allow import/live-reload within angular compiler
```
[the remark this answers is not in the document]
Comment (chrismclarke) [reply tsconfig.app.json:13]: When looking at any file within the IDE it is still just text, and comes down to your IDE whether it thinks comments are OK or not. By default tsconfig files are configured to be assumed `Json with comments` (jsonc)
image

However when it comes to actually parsing/using the json, any interpreter will throw if comments are detected, and so they will have to be stripped out at some point. Thankfully this was done a long time ago for the typescript compiler
`https://github.com/sarod/TypeScript/commit/b60d88fa80bdd09029610584603734419ef89ae9`
